code/modules/mob/living/simple_animal/abnormality/waw/greed_king.dm

```diff
@@ -0,0 +1,207 @@
+	chosen_attack_num = 1
+
+/datum/action/innate/abnormality_attack/kog_teleport
+	name = "Teleport"
+	icon_icon = 'icons/effects/effects.dmi'
+	button_icon_state = "sparks"
+	chosen_message = "<span class='warning'>You will now teleport to a random area in the facility's halls.</span>"
+	chosen_attack_num = 2
+
+/datum/action/innate/abnormality_attack/kog_teleport/Activate()
+	addtimer(CALLBACK(A, .mob/living/simple_animal/hostile/abnormality/greed_king/proc/startTeleport), 1)
+	to_chat(A, chosen_message)
+
+/mob/living/simple_animal/hostile/abnormality/greed_king/Life()
+	. = ..()
+	if(!.) // Dead
+		return FALSE
+	if(!(status_flags & GODMODE))
+		if(!(busy || client))
+			charge_check()
+
+/mob/living/simple_animal/hostile/abnormality/greed_king/AttackingTarget()
+	if(busy)
+		return
+	return ..()
+
+/mob/living/simple_animal/hostile/abnormality/greed_king/Move()
+	if(busy || !client)
+		return FALSE
+	return ..()
+
+/mob/living/simple_animal/hostile/abnormality/greed_king/BreachEffect(mob/living/carbon/human/user)
+	..()
+	icon = 'ModularTegustation/Teguicons/64x48.dmi'
+	//Center it on a hallway
+	pixel_y = -8
+	base_pixel_y = -8
+
+	startTeleport()	//Let's Spaghettioodle out of here
+
+/mob/living/simple_animal/hostile/abnormality/greed_king/proc/startTeleport()
+	if(busy || teleport_cooldown > world.time)
+		return
+	teleport_cooldown = world.time + 4.9 SECONDS
+	//set busy, animate and call the proc that actually teleports.
+	busy = TRUE
+	animate(src, alpha = 0, time = 5)
+	addtimer(CALLBACK(src, .proc/endTeleport), 5)
+
+/mob/living/simple_animal/hostile/abnormality/greed_king/proc/endTeleport()
+	var/turf/T = pick(GLOB.xeno_spawn)
+	animate(src, alpha = 255, time = 5)
+	forceMove(T)
+	busy = FALSE
+	if(!client)
+		addtimer(CALLBACK(src, .proc/startTeleport), 5 SECONDS)
+
+/mob/living/simple_animal/hostile/abnormality/greed_king/proc/charge_check()
+	//targeting
+	var/mob/living/carbon/human/target
+	if(busy)
+		return
+	var/list/possible_targets = list()
+	for(var/mob/living/carbon/human/H in view(20, src))
+		possible_targets += H
+	if(LAZYLEN(possible_targets))
+		target = pick(possible_targets)
+		//Start charge
+		var/dir_to_target = get_cardinal_dir(get_turf(src), get_turf(target))
+		if(dir_to_target)
+			busy = TRUE
+			addtimer(CALLBACK(src, .proc/charge, dir_to_target, 0, target), 2 SECONDS)
+			return
+	return
+
+
+/mob/living/simple_animal/hostile/abnormality/greed_king/OpenFire() // This exists so players can manually charge during playable abnormalities.
+	if(busy || !client)
+		return
+	switch(chosen_attack)
+		if(1)
+			var/dir_to_target = get_cardinal_dir(get_turf(src), get_turf(target))
+			charge(dir_to_target, 0, target)
+	return
+
+/mob/living/simple_animal/hostile/abnormality/greed_king/proc/charge(move_dir, times_ran, target)
+	setDir(move_dir)
+	var/stop_charge = FALSE
+	if(times_ran >= dash_num)
+		stop_charge = TRUE
+	var/turf/T = get_step(get_turf(src), move_dir)
+	if(!T)
```

Comment:
Please call endCharge here so that Busy is unset and she attempt to teleport. Otherwise Busy will never be unset and she will be stuck forever.